front/src/modules/companies/table/components/CompanyTable.tsx

```diff
@@ -32,14 +45,67 @@ export function CompanyTable() {
     return { AND: filters.map(turnFilterIntoWhereClause) };
   }, [filters]) as any;
 
+  const {
+    data: viewFieldsData,
+    loading: isViewFieldsLoading,
+    error: isViewFieldsError,
+  } = useGetViewFieldsQuery({
+    variables: { where: { objectName: { equals: 'company' } } },
+  });
+  const [createViewFieldsMutation] = useCreateViewFieldsMutation();
+
+  useEffect(() => {
```

Comment:
Nice! I think we should:
1) Extract this logic into a hook: useLoadViewFields() (maybe even useLoadView)
2) Move this into a component (maybe into the existing GenericEntityTableData or to a separate GenericEntityTableView). The idea is to keep this computation separate from the parent chains of EntityTable to avoid re-rendering it as this could be expensive

Reply:
Done 👍
I extracted the logic into the `useLoadView` hook which is used by `GenericEntityTableData`.